### PR TITLE
#2009 : Fix hardcoding value forcing parsing fortran as free-form

### PR DESCRIPTION
**Pull request by svalat**
Add optional `free_form` option to parser to handle fixed form fortran.

Feel free to say if you want it another way.

### CODE REVIEW

File: src/psyclone/psyir/frontend/fortran.py [hunk 1]
```diff
@@ -61,10 +61,11 @@ def __init__(self):
         self._processor = Fparser2Reader()
         SYMBOL_TABLES.clear()
 
-    def psyir_from_source(self, source_code):
+    def psyir_from_source(self, source_code, free_form = True):
```
Comment (arporter) [inline src/psyclone/psyir/frontend/fortran.py:64]: Please could you fix the formatting errors reported by pycodestyle for this file.

File: src/psyclone/psyir/frontend/fortran.py [hunk 2]
```diff
@@ -176,12 +177,15 @@ def psyir_from_statement(self, source_code, symbol_table):
...
         return fake_parent[0].detach()
 
-    def psyir_from_file(self, file_path):
+    def psyir_from_file(self, file_path, free_form = True):
         ''' Generate the PSyIR tree representing the given Fortran file.
 
         :param file_path: path of the file to be read and parsed.
         :type file_path: str or any Python Path format.
 
+        :param free_form: If parsing free-form code or not (default False).
```
Comment (arporter) [inline src/psyclone/psyir/frontend/fortran.py:186]: "default True" I think?

File: src/psyclone/tests/psyir/frontend/fortran_test.py [hunk 3]
```diff
@@ -62,6 +62,12 @@
 end subroutine sub
 '''
 
+FIXED_FORM_CODE = '''
```
Comment (arporter) [inline src/psyclone/tests/psyir/frontend/fortran_test.py:65]: Please could you fix the various pycodestyle complaints about this file too.

**Review by arporter — CHANGES_REQUESTED** [review]
I like the fix - thanks very much for doing this PR.
There are just a few pycodestyle errors to fix.
All tests and examples are OK with compilation.
Reference guide builds without new warnings.
Coverage of the diff is 100%.

**Comment by svalat** [comment]
All fixed.

**Review by arporter — APPROVED** [review]
Thanks for making those changes.
All tests and examples are OK with compilation.
Ref. guide builds fine.
Will proceed to merge.

**Pull request merged**